nanoFramework.WebServer/WebServer.cs

```diff
@@ -593,8 +593,15 @@ private void StartListener()
                             context.Response.ContentLength64 = 0;
                         }
 
-                        context.Response.Close();
-                        context.Close();
+                        if (context.Response == null) //When context is handed over to WebsocketServer, this will become null
```

Comment:
```suggestion
                        // When context has been handed over to WebsocketServer, it will be null at this point
                        if (context.Response == null)
```